pkg/sql/schemachanger/scplan/internal/rules/dep_drop_index.go

```diff
@@ -105,3 +105,27 @@ func init() {
 	)
 
 }
+
+func init() {
+	registerDepRuleForDrop(
+		"view can only go to absent when the index becomes hidden",
+		scgraph.SameStagePrecedence,
+		"index", "dependent",
+		scpb.Status_VALIDATED, scpb.Status_ABSENT,
```

Comment:
This should be dropped, I reworked the rules to have saner ordering as suggested.

Reply:
These rules might be very inefficient. Can you file an issue to model the references in the rules engine so that it can be accelerated? In particular, I think this will make drop database cascade for cases where there are many indexes and views O(views * indexes).